apis/apiextensions/v1/composition_types.go

```diff
@@ -197,6 +204,29 @@ type ReadinessCheck struct {
 	MatchInteger int64 `json:"matchInteger,omitempty"`
 }
 
+// Patches that may be applied to a composite or composed resource.
+type Patches []Patch
+
+// Apply a set of patches, optionally filtered by patch type.
+func (p Patches) Apply(composite runtime.Object, composed runtime.Object, only ...PatchType) error {
```

Comment:
The idea here is to lift the filtering of patch types up to the 'array of patches' level, rather than potentially processing the filters on each patch operation.